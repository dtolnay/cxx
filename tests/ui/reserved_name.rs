#[cxx::bridge]
mod ffi {
    struct UniquePtr {
        // invalid; `UniquePtr` is a builtin
        val: usize,
    }

    extern "C++" {
        type Box; // invalid; `Box` is a builtin
    }

    extern "Rust" {
        type String; // valid; `String` is an atom
    }
}

fn main() {}
