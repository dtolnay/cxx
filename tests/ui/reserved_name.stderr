error: reserved name
 --> $DIR/reserved_name.rs:3:12
  |
3 |     struct UniquePtr {
  |            ^^^^^^^^^

error: reserved name
 --> $DIR/reserved_name.rs:9:14
  |
9 |         type Box; // invalid; `Box` is a builtin
  |              ^^^
